cloudscale/resource_cloudscale_server.go

```diff
@@ -406,18 +400,10 @@ func resourceServerDelete(d *schema.ResourceData, meta interface{}) error {
 	log.Printf("[INFO] Deleting Server: %s", d.Id())
 	err := client.Servers.Delete(context.Background(), id)
 
-	if err != nil && strings.Contains(err.Error(), "Not found") {
-		log.Printf("[WARN] Cloudscale Server (%s) not found", d.Id())
-		d.SetId("")
-		return nil
-	}
-
 	if err != nil {
-		return fmt.Errorf("Error deleting Server: %s", err)
+		return CheckDeleted(d, err, "Error deleting Server")
 	}
 
-	d.SetId("")
```

Comment:
I'm not much of a terraform expert, but isn't this supposed to be here?

Reply:
I've been looking through a bunch of other providers (OpenStack, DigitalOcean and AWS) an none set the resource ID to the empty string during delete. 

[Official terraform docs for implementing destroy](https://www.terraform.io/docs/extend/writing-custom-providers.html#implementing-destroy) state:

> d.SetId("") is automatically called assuming delete returns no errors, but
> it is added here for explicitness.

I think we don't have to set it if there is no error.

Reply:
Awesome, thanks!